src/main/java/com/digitalasset/testing/junit4/Sandbox.java

```diff
@@ -113,12 +113,12 @@ public SandboxBuilder logLevel(LogLevel logLevel) {
     public Sandbox build() {
       Objects.requireNonNull(darPath);
 
-      if (useReset && (testModule.isPresent() || testScenario.isPresent())) {
+      if (useReset && (testModule.isPresent() || testStartScript.isPresent())) {
         throw new IllegalStateException(
             "Reset mode cannot be used together with market setup module/scenario.");
```

Comment:
Best solution IMHO would be to add a separate script runner `mySandbox.runScript(String module, String name)`.
And the `reset` could run an init script too.